src/NodaTime/TimeInterval.cs

```diff
@@ -0,0 +1,284 @@
+
+        /// <summary>
+        /// Determines whether the specified <see cref="System.Object" /> is equal to this instance.
+        /// </summary>
+        /// <param name="obj">The <see cref="System.Object" /> to compare with this instance.</param>
+        /// <returns>
+        /// <c>true</c> if the specified <see cref="System.Object" /> is equal to this instance;
+        /// otherwise, <c>false</c>.
+        /// </returns>
+        public override bool Equals(object obj) => obj is TimeInterval other && Equals(other);
+
+        /// <summary>
+        /// Returns the hash code for this instance.
+        /// </summary>
+        /// <returns>
+        /// A 32-bit signed integer that is the hash code for this instance.
+        /// </returns>
+        /// <filterpriority>2</filterpriority>
+        public override int GetHashCode() => HashCodeHelper.Hash(start, end);
+
+        /// <summary>
+        /// Returns a string representation of this TimeInterval, in extended ISO-8601 format: the format
+        /// is "start/end" where each LocalTime uses a format of "uuuu'-'MM'-'dd'T'HH':'mm':'ss;FFFFFFFFF'Z'".
+        /// If the start or end is infinite, the relevant part uses "StartOfTime" or "EndOfTime" to
+        /// represent this.
+        /// </summary>
+        /// <returns>A string representation of this TimeInterval.</returns>
+        public override string ToString()
+        {
+            var pattern = LocalTimePattern.ExtendedIso;
+            return (start.HasValue ? pattern.Format(start.Value) : "-")
+                   + "/"
+                   + (end.HasValue ? pattern.Format(end.Value) : "-");
+        }
+
+        #endregion
+
+        #region Operators
+
+        /// <summary>
+        /// Implements the operator ==.
+        /// </summary>
+        /// <param name="left">The left.</param>
+        /// <param name="right">The right.</param>
+        /// <returns>The result of the operator.</returns>
+        public static bool operator ==(TimeInterval left, TimeInterval right) => left.Equals(right);
+
+        /// <summary>
+        /// Implements the operator !=.
+        /// </summary>
+        /// <param name="left">The left.</param>
+        /// <param name="right">The right.</param>
+        /// <returns>The result of the operator.</returns>
+        public static bool operator !=(TimeInterval left, TimeInterval right) => !(left == right);
+
+        #endregion
+
+        /// <summary>
+        /// Parses the given text value according to the rule LocalTimePattern.ExtendedIso,
+        /// using / char as separator or two LocalTimes
+        /// </summary>
+        /// <param name="text">The text value to parse.</param>
+        /// <returns>The result of parsing.</returns>
+        /// <example>3:44:59/4:00:00</example>
+        /// <example>18:00:00.239/23:00:00</example>
+        /// <exception cref="FormatException"></exception>
+        public static TimeInterval Parse(string text)
```

Comment:
I think I'd prefer to at least make this internal for the moment. We may or may not want a TimeIntervalPattern in the future - it feels odd to have a Parse method on this type but nowhere else. It's easy to make internal methods public later, but we can't go the other way.

Reply:
@jskeet Thank you for the detailed review! I'll fix everything that in my power.
About Parse - I understand that it's not idiomatic way in NodaTime, but we don't have any solution now to parse TimeInterval (I use this method for deserialization in my project). Perhaps we can leave it public but decorate with attribute `[Obsolete("This method is temporary and will be replaced with TimeIntervalPattern.")]`?

Reply:
No, I don't want to have anything "obsolete from the start" in the master - it's too easy to accidentally leave it in there.

The Parse method doesn't rely on anything internal, so you can just have that in your project's code for now, but still use TimeInterval itself in NodaTime.

Reply:
Ok, thank you

Reply:
Removed.